tests/unit/source_code/linters/test_files.py

```diff
@@ -328,44 +347,84 @@ def test_notebook_linter_lints_migrated_table(migration_index, mock_path_lookup)
         end_col=23,
     )
 
+    linter.apply()
+
+    assert path.read_text() == expected
 
-def test_notebook_linter_lints_not_migrated_table(migration_index, mock_path_lookup) -> None:
+
+def test_notebook_linter_lints_and_applies_not_migrated_table(tmp_path, migration_index, mock_path_lookup) -> None:
     """Regression test with the tests above and below."""
-    source = """
+    source_code = """
 # Databricks notebook source
+table_name = 'not_migrated.table'
+
 
-table_name = "not_migrated.table"  # NOT a migrated table according to the migration index
 
 # COMMAND ----------
 
 spark.table(table_name)
+
+
 """.lstrip()
-    linter = _NotebookLinter.from_source_code(migration_index, mock_path_lookup, source, Language.PYTHON)
+    path = tmp_path / "notebook.py"
+    path.write_text(source_code)
+    notebook = Notebook.parse(path, source_code, Language.PYTHON)
+    linter = NotebookLinter(notebook, mock_path_lookup, LinterContext(migration_index))
 
     advices = list(linter.lint())
 
     assert not [advice for advice in advices if advice.code == "table-migrated-to-uc"]
 
+    linter.apply()
+
+    assert path.read_text() == source_code
 
-def test_notebook_linter_lints_migrated_table_with_rename(migration_index, mock_path_lookup) -> None:
+
+def test_notebook_linter_lints_and_applies_migrated_table_with_rename(
```

Comment:
This test fails because the underlying linter does not support fixing the code yet when the value should be inferred instead of being a constant in the table call